### PR TITLE
Use `uuid.uuid4().hex` to generate a random id for sagemaker model names

### LINKED ISSUE #12418: [BUG] Generate Valid AWS Names for Models
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Other

### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### MLflow version

- Client: 2.13.0

### System information

In CircleCI:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 20.04 LTS
- **Python version**: 3.9.7

Locally:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 22.04 LTS
- **Python version**: 3.9.7

### Describe the problem

I have hit an error —two independent times, one from circleci and another time from my local environment— when running `create_deploy` to a sagemaker endpoint as mlflow is trying to create a model in AWS with a random name that AWS does not quite like. This random name shouldn't end in `-` regarding the stacktrace. It's not a major issue as running again `create_deploy` usually works out, but it's kind of annoying.

It feels to me that [this folk](https://github.com/mlflow/mlflow/blob/2248511d5d86bbd751a0fea4241f78be7bc8f3c8/mlflow/sagemaker/__init__.py#L1384) is the one spoiling the party so I guess that just sanitizing the output should do the trick so I'm happy to give it a go unless people more knowledgeable of the platform advise otherwise.

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

Due to the random nature of the function it's hard to reproduce but I guess that one can force the output of `_get_sagemaker_model_name` and try an upload to AWS. This will require having some experiment run in the server, an account in AWS and a valid ECR image. Assuming that all that is there, then it's *as simple* as calling `create_endpoint`:
```python
import os
from mlflow.deployments import get_deploy_client

config = {
    "execution_role_arn": os.getenv("AWS_SAGEMAKER_EXECUTION_ROLE_ARN"),
    "image_url": os.getenv("AWS_ECR_IMAGE"),
    "bucket_name": os.getenv("AWS_S3_ARTIFACT_BUCKET"),
    "instance_type": "ml.m5.large",
    "instance_count": 1,
    "synchronous": True,
}

region = os.getenv("AWS_REGION")
sagemaker = get_deploy_client(f"sagemaker:/{region}")

sagemaker.create_deployment(
    name="some_endpoint_with_a_meaningful_name",
    model_uri="some_model",
    flavor="python_function",
    config=config,
)
```

### Stack trace

```
Traceback (most recent call last):
  File "/home/circleci/project/deploy.py", line 138, in <module>
    callable_(env=REMOTE)
  File "/home/circleci/project/deploy.py", line 113, in __call__
    self._deploy()
  File "/home/circleci/project/deploy.py", line 94, in _deploy
    self.sagemaker.create_deployment(
  File "/home/circleci/project/venv/lib/python3.9/site-packages/******/sagemaker/__init__.py", line 2331, in create_deployment
    app_name, flavor = _deploy(
  File "/home/circleci/project/venv/lib/python3.9/site-packages/******/sagemaker/__init__.py", line 448, in _deploy
    deployment_operation = _create_sagemaker_endpoint(
  File "/home/circleci/project/venv/lib/python3.9/site-packages/******/sagemaker/__init__.py", line 1601, in _create_sagemaker_endpoint
    model_response = _create_sagemaker_model(
  File "/home/circleci/project/venv/lib/python3.9/site-packages/******/sagemaker/__init__.py", line 1899, in _create_sagemaker_model
    return sage_client.create_model(**create_model_args)
  File "/home/circleci/project/venv/lib/python3.9/site-packages/botocore/client.py", line 565, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/circleci/project/venv/lib/python3.9/site-packages/botocore/client.py", line 1021, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the CreateModel operation: 1 validation error detected: Value 'uat-7ae4e0ce66884d02b4487ba45a89f26c-model-cnxhxy9tqbuxajvvvzx-' at 'modelName' failed to satisfy constraint: Member must satisfy regular expression pattern: ^[a-zA-Z0-9]([\-a-zA-Z0-9]*[a-zA-Z0-9])?
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [X] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by harupy**
### Related Issues/PRs

Fix #12418

### What changes are proposed in this pull request?

`get_unique_resource_id` occasionally generates a string ending with `-`, which AWS doesn't accept. To ensure the unique ID ends with an alphanumeric character, use `uuid.uuid4().hex` instead.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/sagemaker/__init__.py [hunk 1]
```diff
@@ -1376,7 +1376,8 @@ def _truncate_name(name, max_length):
 
 
 def _get_unique_name(base_name, unique_suffix, unique_id_length=20):
-    unique_resource_string = f"{unique_suffix}{get_unique_resource_id(unique_id_length)}"
```
Comment (harupy) [inline mlflow/sagemaker/__init__.py:1379]: Another option is to fix `get_unique_resource_id` to only return a string with `[0-9a-z]`.

File: mlflow/sagemaker/__init__.py [hunk 2]
```diff
@@ -32,7 +33,6 @@
 from mlflow.models.model import MLMODEL_FILE_NAME
 from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE, RESOURCE_DOES_NOT_EXIST
 from mlflow.tracking.artifact_utils import _download_artifact_from_uri
-from mlflow.utils import get_unique_resource_id
```
Comment (B-Step62) [inline mlflow/sagemaker/__init__.py:35]: It seems `get_unique_resource_id` is only used for this ID generation, can we remove it?

**Review by B-Step62 — APPROVED** [review]
LGTM, with a minor clean-up suggestion

File: mlflow/sagemaker/__init__.py (hunk 2 above)
Comment (harupy) [reply mlflow/sagemaker/__init__.py:35]: thanks for the catch, it's never used now.

**Pull request merged**